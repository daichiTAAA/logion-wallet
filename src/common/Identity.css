.Identity {
    padding-left: 5px;
    padding-right: 5px;
}

.Identity .IdentityCol {
    padding-left: 10px;
    padding-right: 10px;
}
