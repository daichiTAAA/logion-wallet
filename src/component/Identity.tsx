import React from 'react';
import Form from 'react-bootstrap/Form';
import FormGroup from 'react-bootstrap/FormGroup';

import IdentityType from './types/Identity';

export interface Props {
    identity: IdentityType
}

export default function Identity(props: Props) {
    return (
        <>
            <FormGroup>
                <Form.Label>First name</Form.Label>
                <Form.Control type="text" value={ props.identity.firstName } readOnly />
            </FormGroup>
            <FormGroup>
                <Form.Label>Last name</Form.Label>
                <Form.Control type="text" value={ props.identity.lastName } readOnly />
            </FormGroup>
            <FormGroup>
                <Form.Label>E-mail</Form.Label>
                <Form.Control type="text" value={ props.identity.email } readOnly />
            </FormGroup>
            <FormGroup>
                <Form.Label>Phone number</Form.Label>
                <Form.Control type="text" value={ props.identity.phoneNumber } readOnly />
            </FormGroup>
        </>
    );
}
