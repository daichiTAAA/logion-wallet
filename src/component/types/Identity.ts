export default interface UserIdentity {
    firstName: string,
    lastName: string,
    email: string,
    phoneNumber: string
}
