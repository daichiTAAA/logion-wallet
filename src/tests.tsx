import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
