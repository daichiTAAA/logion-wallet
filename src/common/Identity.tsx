import { Row, Col } from "react-bootstrap";
import { UserIdentity as IdentityType } from "@logion/client";

import { BackgroundAndForegroundColors } from "./ColorTheme";
import ComparableField from "./ComparableField";

import "./Identity.css";

export interface Props {
    identity?: IdentityType
    otherIdentity?: IdentityType
    colors: BackgroundAndForegroundColors
    squeeze: boolean
    noComparison: boolean
}

export default function Identity(props: Props) {

    interface FieldProps {
        id: string
        label: string
        field: (identity: IdentityType) => string
        noComparison: boolean
    }

    function ComparableIdentityField(fieldProps: FieldProps) {
        return (
            <ComparableField
                id={ fieldProps.id }
                label={ fieldProps.label }
                data={ props.identity }
                otherData={ props.otherIdentity }
                field={ fieldProps.field }
                colors={ props.colors }
                squeeze={ props.squeeze }
                noComparison={ props.noComparison }
            />
        );
    }

    return (
        <div className="Identity">
            <Row>
                <Col className="IdentityCol" md={ 6 }>
                    <ComparableIdentityField
                        id="firstName"
                        label="First Name"
                        field={ identity => identity.firstName }
                        noComparison={ props.noComparison }
                    />
                </Col>
                <Col className="IdentityCol" md={ 6 }>
                    <ComparableIdentityField
                        id="lastName"
                        label="Last Name"
                        field={ identity => identity.lastName }
                        noComparison={ props.noComparison }
                    />
                </Col>
            </Row>
            <Row>
                <Col className="IdentityCol" md={ 6 }>
                    <ComparableIdentityField
                        id="email"
                        label="Email"
                        field={ identity => identity.email }
                        noComparison={ props.noComparison }
                    />
                </Col>
                <Col className="IdentityCol" md={ 6 }>
                    <ComparableIdentityField
                        id="phoneNumber"
                        label="Phone Number"
                        field={ identity => identity.phoneNumber }
                        noComparison={ props.noComparison }
                    />
                </Col>
            </Row>
        </div>
    );
}
