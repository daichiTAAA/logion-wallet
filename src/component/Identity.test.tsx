import Identity from './Identity';
import IdentityType from './types/Identity';
import { render } from '../tests';

export const DEFAULT_IDENTITY: IdentityType = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

test("renders", () => {
    const tree = render(<Identity identity={ DEFAULT_IDENTITY } />);
    expect(tree).toMatchSnapshot();
});
