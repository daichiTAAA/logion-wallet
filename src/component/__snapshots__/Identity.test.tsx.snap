// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      First name
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="John"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Last name
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Doe"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      E-mail
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="john.doe@logion.network"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Phone number
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="+1234"
    />
  </div>,
]
`;
